### DISCUSSION TITLE
What happens when you delete environment variables

### DISCUSSION THREAD

**Discussion by JamesSingleton** [vercel/community#3715] [Help] [answered]
### Summary

I setup Sanity with my Next.js application hosted on Vercel before I knew there was an integration in place. However, with some of the newer stuff that has come out with Sanity, it needs some missing env vars. So my question is, what happens if I delete all the Sanity related env vars in Vercel and reconfigure the setup with the Sanity integration? Will my website go down? Or will those env vars be active until a redeploy?

### Steps to Reproduce

N/A

**Comment by amyegan** [comment, marked as answer]
Hey @JamesSingleton! Environment variables remain available on existing deployments. You can verify this by temporarily adding an environment variable, using it in a preview deployment, and then deleting the environment variable in project settings. When you go back to the preview deployment you can see it still uses the environment variable.

**Reply by JamesSingleton** [reply]
@amyegan so just to confirm, I _should_ be able to delete my manually added env vars from Sanity and then setup the integration, validate things work properly on a preview deployment before merging my final changes into my `main` branch.

I just want to make sure I am not going to break what is currently in production.

**Reply by amyegan** [reply]
Given that environment variable changes only take effect when a new deployment or redeployment is made, you should be able to do that without breaking production. However, I always recommend backing up the old values just to be extra sure in case anything goes wrong so you can quickly switch back if needed. Please let me know if you run into any trouble 🙏
